transformer_engine/pytorch/attention.py

```diff
@@ -3140,8 +3200,10 @@ def forward(
             if (seq_offsets_q is None
                 or seq_offsets_k is None
                 or seq_offsets_v is None
-                or seq_offsets_o is None):
+                or seq_offsets_o is None
+                or context_parallel):
                 qkv_group = ''.join([x for x in qkv_layout if x not in 'bst'])
+                qkv_group = 'hd_hd_hd' if context_parallel else qkv_group
```

Comment:
Do we make `q`, `k`, `v` contiguous when `context_parallel=True`? Judging by the `hd_hd_hd` group.

Reply:
yes, I did it [at here](https://github.com/xrennvidia/TransformerEngine/blob/xren/cp_thd/transformer_engine/pytorch/attention.py#L3238-L3239).